pytorch_lightning/trainer/optimizers.py

```diff
@@ -18,13 +18,28 @@
 import torch
 from torch import optim
 from torch.optim.optimizer import Optimizer
+from typing_extensions import Protocol, runtime_checkable
 
 import pytorch_lightning as pl
 from pytorch_lightning.core.optimizer import LightningOptimizer
 from pytorch_lightning.utilities import rank_zero_warn
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
 
 
+@runtime_checkable
+class _SupportedLRScheduler(Protocol):
+    """This class is used to detect if an object is stateful using `isinstance(obj, _SupportedLRScheduler)`"""
+
+    def step(self, *args: Any, **kwargs: Any) -> None:
+        ...
+
+    def state_dict(self) -> Dict[str, Any]:
+        ...
+
+    def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
+        ...
+
```

Comment:
this should go in `core/optimizers.py` https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/core/optimizer.py

given ongoing efforts to remove the trainer optimizer mixin: https://github.com/PyTorchLightning/pytorch-lightning/issues/10681

Reply:
yes I am waiting for that one to merge to make further changes here.